packages/slate/src/changes/at-range.js

```diff
@@ -896,8 +896,27 @@ Changes.insertTextAtRange = (change, range, text, marks, options = {}) => {
   if (normalize !== undefined) {
     normalize = range.isExpanded
   }
+  change.insertTextByKey(key, offset, text, marks, { normalize: false })
 
-  change.insertTextByKey(key, offset, text, marks, { normalize })
+  if (normalize) {
+    // If the commonAncestor of startKey and endKey still exists, or the ancestor
+    // of the commonAncestor still exists, then normalize on the narrowest possible range
+    const commonAncestor = document.getCommonAncestor(startKey, range.endKey)
+    const ancestors = document
+      .getAncestors(commonAncestor.key)
+      .push(commonAncestor)
+    const normalizeAncestor = ancestors.findLast(n =>
+      change.value.document.getDescendant(n.key)
+    )
+    if (normalizeAncestor) {
+      change.normalizeNodeByKey(normalizeAncestor.key)
+      return
+    }
+
+    // If for some strange reason, the commonAncestor of startKey and endKey does not
```

Comment:
When does this "strange reason" occur? Can you change this comment to include `(eg. when ` or whatever explaining when this might be the case.

Reply:
(Or actually can you just add a single comment that explains this whole piece of logic clearly and concisely.)

Reply:
@ianstormtaylor I am not sure about one thing: If the document includes one block and the block is void.  Will the document.key be regenerated after deleting the void block?

Reply:
I read about the `deleteAtRange`.  The document.key is always in the ancestors and should be unchanged even with the scenario of select-all(document has only only void blocks) and delete all. Then I delete that line and fix with comment